src/cargo/util/toml/mod.rs

```diff
@@ -3146,6 +3157,120 @@ impl<'de> de::Deserialize<'de> for TomlDebugInfo {
     }
 }
 
+#[derive(Clone, Debug, PartialEq, Eq, Ord, PartialOrd, Hash, Serialize)]
+#[serde(untagged)]
+pub enum TomlTrimPaths {
+    One(TomlTrimPathsValue),
+    Many(Vec<TomlTrimPathsValue>),
```

Comment:
Do we need to preserve `One` vs `Many` or can we just have the deserialize always put it in a `Vec`?

Reply:
My thought is modeling what we see in user-end, but might be overkill you're right.

Reply:
My thought is mostly around keeping the internals simple rather than having multiple ways of expressing the same thing

Reply:
Looking at this more, I think I would personally model this as
```rust
enum TomlTrimPaths {
    None,
    Values(Vec<TomlTrimPathValues>),
    All,
}

enum TomlTrimPathValues {
    Objects,
    Diagnostics,
    Macro,
}
```

Reply:
I went with the second one you suggested. Thanks for the idea and review!